### ISSUE TITLE
Bugs when there are two consecutive get_local_id

### ISSUE THREAD

**Issue by drcut** [KhronosGroup/SPIRV-LLVM-Translator#978]
Hi, I have tried to support converting NVPTX into SPIRV and then convert into LLVM IR. And when I convert the following code
`int row = threadIdx.y;`
, it can generate the corret LLVM bitcode
`%0 = call spir_func i64 @_Z12get_local_idj(i32 1) #1`

However, when the input code is 
`int row = threadIdx.y;
int col = threadIdx.x;`
The generated LLVM will be 
`%0 = call spir_func i64 @_Z12get_local_idj(i32 1)`
`%2 = load <3 x i64>, <3 x i64> addrspace(5)* @__spirv_BuiltInLocalInvocationId1`
`%3 = extractelement <3 x i64> %2, i32 0 `
So I think when converting a SPIRV file into LLVM IR, when there are two consecutive get_local_id, it will raise error.
Would you please give me some suggestions?
Thanks in advance!

**Comment by AlexeySachkov** [comment]
Hi @drcut,

> Would you please give me some suggestions?

Unfortunately, I can't suggest anything right now, but having that SPIR-V file which triggers this issue would definitely help

**Comment by drcut** [comment]
Thanks, here are the links of my spv file
[naive_matmul.spv](https://drive.google.com/file/d/1vEqgJInlfNHOixPkiKGQfHKLOmxGeQ1T/view?usp=sharing).
I also put its corresponding cuda file, as it may be helpful.
[naive_matmul.cu](https://drive.google.com/file/d/1C7m1Djcme-h_tuWx3P6AiQlNTRLhbB1M/view?usp=sharing)

**Comment by drcut** [comment]
@AlexeySachkov Do you have any suggestions for why there is an error when there is more than one get_local_idj? I'd like to study this issue recently, and want to get some hints from you. Thanks in advance

**Comment by drcut** [comment]
So I used a quite ugly way to solve the bug. I create some variable (in the below code, the `LocalInvocationId`) to store the generated GlobalVariable, so that each time we meet a built-in function, we first check whether its corresponding GlobalVariable existing, if not, we will create a new one, else, we directly use the previous generated GlobalVariable
```cpp
    if (BuiltinVarName == "__spirv_BuiltInLocalInvocationId") {
      if (!LocalInvocationId)
        LocalInvocationId = new GlobalVariable(
            *M, GVType, true, GlobalValue::ExternalLinkage, nullptr,
            BuiltinVarName, 0, GlobalVariable::NotThreadLocal, SPIRAS_Input);
      BV = LocalInvocationId;
    }
    // auto BV = new GlobalVariable(*M, GVType, true,
    // GlobalValue::ExternalLinkage,
    //                              nullptr, BuiltinVarName, 0,
    //                              GlobalVariable::NotThreadLocal,
    //                              SPIRAS_Input);
```

**Comment by AlexeySachkov** [comment]
@drcut, good catch. I think the right way to fix this would be using [`Module::getGlobalVariable`](https://llvm.org/doxygen/classllvm_1_1Module.html#aeb548ad4e336f7f3d3cc91578b5de0a8) to check if such global already exists before creating a new one.

I wonder why we haven't found this by existing OpenCL/SYCL tests that go through the SPIR-V

**Comment by AlexeySachkov** [comment]
I suggest that we leave this open, because the issue hasn't been fixed
